src/runtime/nitro/plugins/40-cspSsrNonce.ts

```diff
@@ -0,0 +1,57 @@
+import { defineNitroPlugin } from '#imports'
+import crypto from 'node:crypto'
+import { resolveSecurityRules } from '../context'
+
+const LINK_RE = /<link([^>]*?>)/g
+const SCRIPT_RE = /<script([^>]*?>)/g
```

Comment:
@GalacticHypernova do you know how we could modify the script regex to also match upper case `<SCRIPT>` tags?

Reply:
@vejja Is there a reason for that? Theoretically we can add the i flag for case insensitivity, but who would make script tags uppercase?

Reply:
hi @GalacticHypernova 
This is a CodeQL warning, it is related to several CVE issues.
It prevents the CI workflow to complete.
Full details given by CodeQL output here : https://github.com/Baroshem/nuxt-security/security/code-scanning/3

Reply:
I mean, sure, let's add the /i flag so it would be /ig

Reply:
Ok thanks
I'm not entirely sure what the underlying issue is to be honest, but here is a screenshot
<img width="1134" alt="Screenshot 2024-05-17 at 16 27 32" src="https://github.com/Baroshem/nuxt-security/assets/7295259/727e8d8e-6fd7-4ad6-8b07-d5f776703faf">

Reply:
That's actually a great example, I didn't know script tags can have attrs at the closing tag. That means the regex needs to be reworked from `</script>` to `</script[^>]*?>` to avoid that edge case. The rest of the edge cases I believe is already taken care of with the pattern itself

Reply:
Also, regarding the candidates for the optional `as`, I believe it's a good choice to stay with the syntax you suggested, as it makes the pattern more concise and avoids the out-of-place trailing `|`. They don't have any noticeable performance differences either, so that's good.